tests/test_everflow_testbed.py

```diff
@@ -0,0 +1,368 @@
+
+    tor_ports = []
+    spine_ports = []
+    port_channels = []
+    everflow_table_ports = []
+
+    # gather ansible facts
+    mg_facts = duthost.minigraph_facts(host=duthost.hostname)['ansible_facts']
+    #gather switch capability facts
+    switch_capability_facts = duthost.switch_capabilities_facts()['ansible_facts']
+    #gather host facts
+    host_facts = duthost.setup()['ansible_facts']
+
+    # get the list of TOR/SPINE ports
+    for dut_port, neigh in mg_facts['minigraph_neighbors'].items():
+        if 'T0' in neigh['name']:
+            tor_ports.append(dut_port)
+        elif 'T2' in neigh['name']:
+            spine_ports.append(dut_port)
+
+    # get the list of port channels
+    port_channels = mg_facts['minigraph_portchannels']
+
+    # get the list of port to be combined to ACL tables
+    everflow_table_ports += tor_ports
+    if testbed['topo']['name'] in ('t1-lag', 't1-64-lag', 't1-64-lag-clet'):
+        everflow_table_ports += port_channels
+    else:
+        everflow_table_ports += spine_ports
+
+    test_mirror_v4 = True if switch_capability_facts['switch_capabilities']['switch']['MIRROR'] == 'true' else False
+    test_mirror_v6 = True if switch_capability_facts['switch_capabilities']['switch']['MIRRORV6'] == 'true' else False
+    test_ingress_mirror_on_ingress_acl = True if 'MIRROR_INGRESS_ACTION' in switch_capability_facts['switch_capabilities']['switch']['ACL_ACTIONS|INGRESS'] else False
+    test_ingress_mirror_on_egress_acl =  True if 'MIRROR_INGRESS_ACTION' in switch_capability_facts['switch_capabilities']['switch']['ACL_ACTIONS|EGRESS'] else False
+    test_egress_mirror_on_egress_acl =  True if 'MIRROR_EGRESS_ACTION' in switch_capability_facts['switch_capabilities']['switch']['ACL_ACTIONS|EGRESS'] else False
+    test_egress_mirror_on_ingress_acl = True if 'MIRROR_EGRESS_ACTION' in switch_capability_facts['switch_capabilities']['switch']['ACL_ACTIONS|INGRESS'] else False
+
+
+    def get_port_info(in_port_list, out_port_list, out_port_ptf_id_list, out_port_lag_name): 
+        ptf_port_id = ''
+        out_port_exclude_list = []
+        for port in in_port_list:
+            if port not in out_port_list and port not in out_port_exclude_list and len(out_port_list) < 3:
+                ptf_port_id += (str(mg_facts['minigraph_port_indices'][port]))
+                out_port_list.append(port)
+                out_port_lag_name.append("Not Applicable")
+                for portchannelinfo in mg_facts['minigraph_portchannels'].items():
+                    if port in portchannelinfo[1]['members']:
+                        out_port_lag_name[-1] = portchannelinfo[0]
+                        for lag_memeber in portchannelinfo[1]['members']:
+                            if port == lag_memeber:
+                                continue
+                            ptf_port_id += "," +  (str(mg_facts['minigraph_port_indices'][lag_memeber]))
+                            out_port_exclude_list.append(lag_memeber)
```

Comment:
When adding a port to exclude list, should you also try to remove it from out_port_list in case it has been added?

Or should the exclude list be calculated before entering this loop?

Reply:
@yxieca port will not get added to out_port_list. Before adding port to out_port_list we are making check it's not part of exclude_list and also existing port_list. 

Idea we want to get unique set of ports in case across lag

Reply:
But you are calculating the exclude_list in the same loop as adding the port? So the port could be added then later added to exclude list. I don't see your code avoid this possibility? Did I miss something?

Reply:
@yxieca :  following check make sure that we don't add port to exclude list if it's already added to out_port_list
                if port == lag_member:
                        continue
we are doing following:
a) Walk through in_port_list
b) Add port to out_port_list if not already in out_port_list and not in out_port_exclude_list
c) If added port is part of lag member then except that port added all other remaining port in exclude list